GitTfsTest/Integration/CloneTests.cs

```diff
@@ -324,5 +325,61 @@ public void HandlesIgnoredFilesParticipatingInRenames()
         }
 
         #endregion
+
+        [FactExceptOnUnix]
+        public void CloneWithAllBranchesShouldHandleFolderDeletedAndRecreatedAsBranch()
+        {
+            //Arrange
+            h.SetupFake(
+                r =>
+                {
+                    r.SetRootBranch("$/MyTeamProject/Root");
+                    r.Changeset(1, "Create initial team project.", DateTime.Now)
+                        .Change(TfsChangeType.Add, TfsItemType.Folder, "$/MyTeamProject");
+
+                    r.Changeset(2, "Create root branch.", DateTime.Now)
+                        .Change(TfsChangeType.Add, TfsItemType.Folder, "$/MyTeamProject/Root");
+
+                    r.Changeset(3, "Create \"branch\" (as a folder).", DateTime.Now)
+                        .Change(TfsChangeType.Add, TfsItemType.Folder, "$/MyTeamProject/Branch");
+
+                    r.Changeset(4, "Delete the \"branch\" folder.", DateTime.Now)
+                        .Change(TfsChangeType.Delete, TfsItemType.Folder, "$/MyTeamProject/Branch");
+
+                    r.BranchChangeset(5, "Create a proper branch.", DateTime.Now, "$/MyTeamProject/Root", "$/MyTeamProject/Branch", 2)
+                        .Change(TfsChangeType.Branch, TfsItemType.Folder, "$/MyTeamProject/Branch");
+                });
+
+            //Act
+            h.Run("clone", h.TfsUrl, "$/MyTeamProject/Root", "MyTeamProject", "--branches=all");
+
+            //Assert
+            h.AssertGitRepo("MyTeamProject");
+
+            var branchCollection = h.Repository("MyTeamProject").Branches.Cast<Branch>().ToList();
+            var branch = branchCollection.FirstOrDefault(b => b.Name == "Branch");
+
+            //Ensure our Branch was migrated...
+            Assert.NotNull(branch);
+
+            // So, it turns out GetRootChangesetForBranch is really the unit under test here.
```

Comment:
Hmm, maybe you could create a [Test Spy](http://xunitpatterns.com/Test%20Spy.html) or some other kind of ["test double"](http://xunitpatterns.com/Test%20Double.html) to help you assert/verify the behavior you're expecting?